### PR TITLE
Key Metrics Dashboard

### PR DESCRIPTION
**Pull request by marcus-snx**
Work In Progress!

Use `streamlit run dashboards/dash_key_metrics/key_metrics_app.py` to run the dashboards service locally.

### CODE REVIEW

File: dashboards/dash_key_metrics/key_metrics_app.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    return SynthetixAPI(db_config=get_db_config(streamlit=True))
+
+
+st.session_state.api = load_api()
+
+# pages
+cross_chain = st.Page("views/cross_chain.py", title="Cross-chain stats")
+lp = st.Page("views/lp.py", title="Liquidity Providers")
+perps = st.Page("views/perps.py", title="Perps")
+token = st.Page("views/token.py", title="Token")
```
Comment (Tburm) [inline dashboards/dash_key_metrics/key_metrics_app.py:30]:
```suggestion
token = st.Page("views/token.py", title="$SNX Token")
```

I think something that specifically mentions SNX is better here.

File: dashboards/dash_key_metrics/key_metrics_app.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# set the API
+@st.cache_resource
+def load_api():
+    return SynthetixAPI(db_config=get_db_config(streamlit=True))
+
+
+st.session_state.api = load_api()
+
+# pages
+cross_chain = st.Page("views/cross_chain.py", title="Cross-chain stats")
```
Comment (Tburm) [inline dashboards/dash_key_metrics/key_metrics_app.py:27]: I don't have a good suggestion, but I don't think "Cross-chain stats" is the best title here. I think for a landing page we might just want to name something it "Synthetix V3"

File: dashboards/dash_key_metrics/views/lp.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+data = fetch_data(st.session_state.date_range, st.session_state.chain)
+
+chart_core_tvl_by_collateral = chart_area(
+    data["core_stats_by_collateral"],
+    x_col="ts",
+    y_cols="collateral_value",
+    title="TVL",
+    color="label",
+)
+chart_core_apr_by_collateral = chart_area(
```
Comment (Tburm) [inline dashboards/dash_key_metrics/views/lp.py:66]: This should be a line chart

File: dashboards/dash_key_metrics/views/lp.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    data["core_stats_by_collateral"],
+    x_col="ts",
+    y_cols="apr",
+    title="APR",
+    color="label",
+)
+chart_core_apr_rewards_by_collateral = chart_area(
```
Comment (Tburm) [inline dashboards/dash_key_metrics/views/lp.py:73]: This should be a line chart

File: dashboards/dash_key_metrics/views/lp.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    color="label",
+)
+chart_core_debt_by_collateral = chart_area(
+    data["core_stats_by_collateral"],
+    x_col="ts",
+    y_cols="debt",
+    title="Debt",
+    color="label",
+)
+chart_core_rewards_usd_by_collateral = chart_area(
```
Comment (Tburm) [inline dashboards/dash_key_metrics/views/lp.py:87]: These should be stacked bars

File: dashboards/dash_key_metrics/views/lp.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    color="label",
+)
+chart_core_apr_rewards_by_collateral = chart_area(
+    data["core_stats_by_collateral"],
+    x_col="ts",
+    y_cols="apr_rewards",
+    title="APR (Rewards)",
+    color="label",
+)
+chart_core_debt_by_collateral = chart_area(
```
Comment (Tburm) [inline dashboards/dash_key_metrics/views/lp.py:80]: Now that I'm seeing this, I really wonder if this chart should exist at all... I'm not sure it's very relevant for a non-technical or general user.

**Review by Tburm — COMMENTED** [review]
Leaving some naming and formatting comments. Not all need to be addressed before we merge this, I'm thinking we should do a full review of the copy with product/design before any rollout to match it to other communication.

File: api/internal_api.py [hunk 7]
```diff
@@ -108,3 +169,152 @@ def get_volume(
         """
         with self.get_connection() as conn:
             return pd.read_sql_query(query, conn)
+
+    def get_core_stats_by_chain(
```
[the remark this answers is not in the document]
Comment (marcus-snx) [reply api/internal_api.py:173]: 100% agree. We should also probably have a look at the way the views/tables are named on the data service side, e.g. the apr tables hold a lot of info beyond apr.

**Pull request merged**